ownership.go

```diff
@@ -20,35 +20,56 @@ import (
 	"os"
 	"strconv"
 	"strings"
-)
 
-var passwdFile = "/etc/passwd"
-var groupFile = "/etc/group"
+	"github.com/Sirupsen/logrus"
+)
 
 // Ownership indicates the default ownership of filesystem entries.
 type Ownership struct {
 	UID uint32
 	GID uint32
+	// Where to look up users and groups
+	userSource  string
+	groupSource string
 }
 
 // NewOwnership initializes default file ownership struct.
-func NewOwnership(username, groupname string) (Ownership, error) {
-	uid, err := lookupUID(username)
+// Logs as error anything that goes wrong, but always returns something
+// Worst-case you get "0", ie root, owning things, which is safe as root can always read all files.
+func NewOwnership(username, groupname, fallbackUser, fallbackGroup, passwdFile, groupFile string, logger *logrus.Entry) Ownership {
+	var uid, gid uint32
+	var err error
+
+	uid, err = lookupUID(username, passwdFile)
 	if err != nil {
-		return Ownership{}, err
+		logger.WithError(err).WithField("user", username).Error("Error looking up username, using fallback")
+		uid, err = lookupUID(fallbackUser, passwdFile)
+		if err != nil {
+			uid = 0
+			logger.WithError(err).WithField("user", username).Error("Error looking up fallback username, using 0")
+		}
 	}
-	gid, err := lookupGID(groupname)
+
+	gid, err = lookupGID(groupname, groupFile)
 	if err != nil {
-		return Ownership{}, err
+		logger.WithError(err).WithField("group", groupname).Error("Error looking up groupname, using fallback")
+		gid, err = lookupGID(fallbackGroup, groupFile)
+		if err != nil {
+			gid = 0
+			logger.WithError(err).WithField("group", groupname).Error("Error looking up fallback groupname, using 0")
+		}
 	}
+
 	return Ownership{
-		UID: uid,
-		GID: gid,
-	}, nil
+		UID:         uid,
+		GID:         gid,
+		userSource:  passwdFile,
+		groupSource: groupFile,
+	}
 }
 
 // lookupUID resolves a username to a numeric id.
-func lookupUID(username string) (uint32, error) {
+func lookupUID(username, passwdFile string) (uint32, error) {
 	uid, err := lookupIDInFile(username, passwdFile)
 	if err != nil {
 		return 0, fmt.Errorf("Error resolving uid for %s: %v\n", username, err)
```

Comment:
Add the filename to `fmt.Errorf("Error resolving uid for %s: %v\n", username, err)` as well as to the final error message?